### PR TITLE
feat: adopt Grafana for metrics visualization (#2364)

### PR DESCRIPTION
**Pull request by jonathanj-square**
fixes: #2364 

Introduces a Grafana / OTEL container for capturing, ephemerally storing, and visualizing OTEL metrics emitted by FTL. The collection frequency is also increased to once per second (from a default once per minute)

Screenshot 2024-08-14 at 7 58 39 PM

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+OTEL_GRPC_PORT=4317
+OTEL_HTTP_PORT=4318
```
Comment (alecthomas) [inline .env:2]: Use Hermit for managing these envars instead:

```
hermit env OTEL_GRPC_PORT 4317
hermit env OTEL_HTTP_PORT 4318
```

**Review by alecthomas — APPROVED** [review]

**Comment by alecthomas** [comment]
@jonathanj-square I pulled it out of the queue because there are a couple of changes that would be nice to make

File: docker-compose.yml [hunk 2]
```diff
@@ -18,6 +18,20 @@ services:
       timeout: 60s
       retries: 60
       start_period: 80s
+  grafana:
+    image: grafana/otel-lgtm
+    platform: linux/amd64
+    restart: unless-stopped
```
[the remark this answers is not in the document]
Comment (jonathanj-square) [reply docker-compose.yml:26]: done

File: Justfile [hunk 3]
```diff
@@ -223,26 +205,39 @@ otel-stream:
...
-    -p {{otelGrpcPort}}:{{otelGrpcPort}} \
+    -p {{otel-grpc-port}}:{{otel-grpc-port}} \
     -p 55679:55679 \
     otel/opentelemetry-collector:0.104.0 2>&1 | sed 's/\([A-Z].* #\)/\
   \1/g'
 
 # Stop the docker container running otel.
-otelContainerID := `docker ps -f ancestor=otel/opentelemetry-collector:0.104.0 | tail -1 | cut -d " " -f1`
+otel-container-id := `docker ps -f ancestor=otel/opentelemetry-collector:0.104.0 | tail -1 | cut -d " " -f1`
+grafana-container-id := `docker ps -f ancestor=grafana/otel-lgtm | tail -1 | cut -d " " -f1`
```
[the remark this answers is not in the document]
Comment (jonathanj-square) [reply Justfile]: fixed

**Pull request merged**